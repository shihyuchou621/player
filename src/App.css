.player {
  left: 50%;
  position: absolute;
  transform: translate(-50%, 0);
  padding: 30px;
}

.songArea {
  width: 420px;
  height: 600px;
}

.controlPanel {
  display: flex;
}

.controlPanel > *:nth-child(1) {
  flex: 1;
}

.controlPanel > *:nth-child(2) {
  flex: 2;
}

.controlPanel:after {
  content: '';
  flex: 1;
}