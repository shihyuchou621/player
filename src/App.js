import './App.css';
import { createStore } from "redux";

import { Status, Mode } from "./consts";
import {
  toggleStar,
  none,
  play,
  pause,
  resume,
  ticking,
  modeChange,
} from './actions';
import reducers from "./reducers";

const store = createStore(
  reducers,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

store.dispatch(play(1));

store.dispatch(pause());

store.dispatch(resume());

store.dispatch(ticking());

store.dispatch(none());


store.dispatch(toggleStar(2));

store.dispatch(
  modeChange({mode: Mode.ALL})
);



// console.log(store.getState());

export default function App() {
  return (
    <div className="App">
      player
    </div>
  );
}