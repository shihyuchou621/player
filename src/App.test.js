import { render, screen } from '@testing-library/react';
import App from './App';

test('rbg should be correct', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});