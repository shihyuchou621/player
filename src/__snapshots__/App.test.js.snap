// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rbg should be correct 1`] = `
<div>
  <div
    class="player"
  >
    <div
      class="songArea"
    >
      <div
        class="song playButton"
      >
        <div
          class="button current playSign sign"
        />
        <div
          class="starButton"
          data-star="☆"
        />
        <div
          class="songsData"
        >
          <div
            class="songData"
          >
             
            song1
             
          </div>
          <div
            class="songData"
          >
             
            artist1
             
          </div>
        </div>
      </div>
      <div
        class="song playButton"
      >
        <div
          class="button playSign sign"
        />
        <div
          class="starButton"
          data-star="☆"
        />
        <div
          class="songsData"
        >
          <div
            class="songData"
          >
             
            song2
             
          </div>
          <div
            class="songData"
          >
             
            artist2
             
          </div>
        </div>
      </div>
      <div
        class="song playButton"
      >
        <div
          class="button playSign sign"
        />
        <div
          class="starButton"
          data-star="☆"
        />
        <div
          class="songsData"
        >
          <div
            class="songData"
          >
             
            song3
             
          </div>
          <div
            class="songData"
          >
             
            artist3
             
          </div>
        </div>
      </div>
      <div
        class="song playButton"
      >
        <div
          class="button playSign sign"
        />
        <div
          class="starButton"
          data-star="☆"
        />
        <div
          class="songsData"
        >
          <div
            class="songData"
          >
             
            song4
             
          </div>
          <div
            class="songData"
          >
             
            artist4
             
          </div>
        </div>
      </div>
      <div
        class="song playButton"
      >
        <div
          class="button playSign sign"
        />
        <div
          class="starButton"
          data-star="☆"
        />
        <div
          class="songsData"
        >
          <div
            class="songData"
          >
             
            song5
             
          </div>
          <div
            class="songData"
          >
             
            artist5
             
          </div>
        </div>
      </div>
      <div
        class="song playButton"
      >
        <div
          class="button playSign sign"
        />
        <div
          class="starButton"
          data-star="☆"
        />
        <div
          class="songsData"
        >
          <div
            class="songData"
          >
             
            song6
             
          </div>
          <div
            class="songData"
          >
             
            artist6
             
          </div>
        </div>
      </div>
      <div
        class="song playButton"
      >
        <div
          class="button playSign sign"
        />
        <div
          class="starButton"
          data-star="☆"
        />
        <div
          class="songsData"
        >
          <div
            class="songData"
          >
             
            song7
             
          </div>
          <div
            class="songData"
          >
             
            artist7
             
          </div>
        </div>
      </div>
      <div
        class="song playButton"
      >
        <div
          class="button playSign sign"
        />
        <div
          class="starButton"
          data-star="☆"
        />
        <div
          class="songsData"
        >
          <div
            class="songData"
          >
             
            song8
             
          </div>
          <div
            class="songData"
          >
             
            artist8
             
          </div>
        </div>
      </div>
      <div
        class="song playButton"
      >
        <div
          class="button playSign sign"
        />
        <div
          class="starButton"
          data-star="☆"
        />
        <div
          class="songsData"
        >
          <div
            class="songData"
          >
             
            song9
             
          </div>
          <div
            class="songData"
          >
             
            artist9
             
          </div>
        </div>
      </div>
    </div>
    <div
      class="SongBar"
    >
      <div
        class="bar"
      >
        <div
          class="progress"
          style="width: 0%;"
        />
      </div>
      <div
        class="passTime"
      >
        0:00
      </div>
      <div
        class="lastTime"
      >
        -3:00
      </div>
    </div>
    <div
      class="controlPanel"
    >
      <div>
        <button>
           
          ONE
           
        </button>
      </div>
      <div
        class="Controller"
      >
        <div
          class="lastSong bigSign"
        />
        <div
          class="bigPlaySign bigSign"
        />
        <div
          class="nextSong bigSign"
        />
      </div>
    </div>
  </div>
</div>
`;
